### PR TITLE
fixes #28973 - Add disable option for user

### PR DESCRIPTION
**Pull request by domitea**
Hammer part of https://github.com/theforeman/foreman/pull/7402

### CODE REVIEW

File: lib/hammer_cli_foreman/param_filters.rb [hunk 1]
```diff
@@ -25,6 +25,7 @@ def flatten_params(params)
       params.each do |p|
         result += flatten_params(p.params)
       end
+      puts result
```
Comment (shiramax) [inline lib/hammer_cli_foreman/param_filters.rb]: Is there a reason to print the result?

Comment (ofedoren) [reply lib/hammer_cli_foreman/param_filters.rb]: I think this is just debug leftover.

File: lib/hammer_cli_foreman/user.rb [hunk 2]
```diff
@@ -57,6 +58,8 @@ def self.included(base)
         base.option '--default-location', 'DEFAULT_LOCATION_NAME', _("Default location name")
         base.option "--ask-password", "ASK_PW", " ",
                     :format => HammerCLI::Options::Normalizers::Bool.new
+        base.option '--disabled', 'USER_DISABLED', ' ',
+                    :format => HammerCLI::Options::Normalizers::Bool.new
```
Comment (ofedoren) [inline lib/hammer_cli_foreman/user.rb]: Actually this is not needed unless there is no such param in API doc or you want to apply custom changes before sending the request.

The `--disabled` option will be created automatically after `hammer -r` if apidoc cache was updated.

File: lib/hammer_cli_foreman/user.rb [hunk 3]
```diff
@@ -82,6 +85,12 @@ class UpdateCommand < HammerCLIForeman::UpdateCommand
       include CommonUpdateOptions
 
       build_options
+
+      def request_params
+        params = super
+        params['user']['disabled'] = option_disabled
+        params
+      end
```
Comment (ofedoren) [inline lib/hammer_cli_foreman/user.rb]: This is not needed.

File: lib/hammer_cli_foreman/param_filters.rb (hunk 1 above)
Comment (domitea) [reply lib/hammer_cli_foreman/param_filters.rb]: Yes, It was a leftover. Thanks!

**Comment by domitea** [comment]
@ofedoren @shiramax Code improved, looking good for you?

**Comment by ofedoren** [comment]
ok to test

**Comment by shiramax** [comment]
no, thank you @domcleal  and @ofedoren, merging :)

**Pull request merged**